Topic 1/src/point6/AbstrsctProduct.java

```diff
@@ -0,0 +1,68 @@
+package point6;
+
+public abstract class AbstrsctProduct {
+	
+	private int id;
+	private String name;
+	private double price;
+	private int units;
+	private MailList mailList;
+	
+	public AbstrsctProduct(int id, String name, double price, int units, MailList mailList){
+		this.id= id;
+		this.name= name;
+		this.price= price;
+		this.units= units;
+		this.mailList= mailList;
+	}
+	
+	public abstract String getDescription();
+	
+	public double subTotalPrice(){
+		return (this.getPrice()* this.getUnits());
+	}
+
+	public String getName() {
+		return name;
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+
+	public double getPrice() {
+		return price;
+	}
+
+	public void setPrice(double price) { //Notify
+		this.price = price;
+		this.mailList.SendEmail("This price was chanched", this.getDescription());
```

Comment:
[High] Instead of refering to *Observer* methods, you can only notify an event, and then each observer can treat them as they wish. You can read more about *Observer* pattern